doc/tutorials/pick_and_place_with_moveit_task_constructor/pick_and_place_with_moveit_task_constructor.rst

```diff
@@ -0,0 +1,899 @@
+
+    static const rclcpp::Logger LOGGER = rclcpp::get_logger("mtc_tutorial");
+    namespace mtc = moveit::task_constructor;
+
+We start by defining a class that will contain the main MoveIt Task Constructor functionality.  We also declare the MoveIt Task Constructor task object as a member variable for our class: this isn't strictly necessary for a given application, but it helps save the task for later visualization purposes. We will explore each function individually below.
+
+.. code-block:: c++
+
+    class MTCTaskNode
+    {
+    public:
+      MTCTaskNode(const rclcpp::NodeOptions& options);
+
+      rclcpp::node_interfaces::NodeBaseInterface::SharedPtr getNodeBaseInterface();
+
+      void doTask();
+
+      void setupPlanningScene();
+
+    private:
+      // Compose an MTC task from a series of stages.
+      mtc::Task createTask();
+      mtc::Task task_;
+      rclcpp::Node::SharedPtr node_;
+    };
+
+These lines define a getter function to get the node base interface, which will be used for the executor later.
+
+.. code-block:: c++
+
+    rclcpp::node_interfaces::NodeBaseInterface::SharedPtr MTCTaskNode::getNodeBaseInterface()
+    {
+      return node_->get_node_base_interface();
+    }
+
+These next lines initialize the node with specified options.
+
+.. code-block:: c++
+
+    MTCTaskNode::MTCTaskNode(const rclcpp::NodeOptions& options)
+      : node_{ std::make_shared<rclcpp::Node>("mtc_node", options) }
+    {
+    }
+
+This class method is used to set up the planning scene that is used in the example. It creates a cylinder with dimensions specified by ``object.primitives[0].dimensions`` and position specified by ``pose.position.z`` and ``pose.position.x``. You can try changing these numbers to resize and move the cylinder around. If you move the cylinder out of the robot's reach, planning will fail.
+
+.. code-block:: c++
+
+    void MTCTaskNode::setupPlanningScene()
+    {
+      moveit_msgs::msg::CollisionObject object;
+      object.id = "object";
+      object.header.frame_id = "world";
+      object.primitives.resize(1);
+      object.primitives[0].type = shape_msgs::msg::SolidPrimitive::CYLINDER;
+      object.primitives[0].dimensions = { 0.1, 0.02 };
+
+      geometry_msgs::msg::Pose pose;
+      pose.position.x = 0.5;
+      pose.position.y = -0.25;
+      object.pose = pose;
+
+      moveit::planning_interface::PlanningSceneInterface psi;
+      psi.applyCollisionObject(object);
+    }
+
+This function interfaces with the MoveIt Task Constructor task object. It first creates a task, which includes setting some properties and adding stages. This will be discussed further in the ``createTask`` function definition. Next, ``task.init()`` initializes the task and ``task.plan(5)`` generates a plan, stopping after 5 successful plans are found. The next line publishes the solution to be visualized in RViz - this line can be removed if you don't care for visualization. Finally, ``task.execute()`` executes the plan.
```

Comment:
```suggestion
This function interfaces with the MoveIt Task Constructor task object. It first creates a task, which includes setting some properties and adding stages. This will be discussed further in the ``createTask`` function definition. Next, ``task.init()`` initializes the task and ``task.plan(5)`` generates a plan, stopping after 5 successful plans are found. The next line publishes the solution to be visualized in RViz - this line can be removed if you don't care for visualization. Finally, ``task.execute()`` executes the plan. Execution occurs via an action server interface with the RViz plugin.
```